### DISCUSSION TITLE
Can't upload files larger than 500KB into a database storage

### DISCUSSION THREAD

**Discussion by estevez-dev** [serverpod/serverpod#2617] [Q&A] [answered]
Hi. I'm trying to implement image upload according to the documentation and I can't upload images larger than 500KB approximately. There are no errors on the client or server side. The upload finishes successfully, but `verifyDirectFileUpload` returns `false`.

Client upload code:
```dart
final uploadDescription =
    await client.myEndpoint.getImageUploadDescription(
  fileName: fileName,
);

// Upload description exists for both, large and small images
if (uploadDescription != null) {
  final uploader = FileUploader(uploadDescription);

  // `data` here has `Uint8List` type
  final uploaded = await uploader.uploadByteData(
    ByteData.view(data.buffer),
  );

  //`uploaded` is `true` for both, large and small images 
  if (!uploaded) {
    return const Failure();
  }

  // Here I'm getting `true` from images smaller than 500KB, and `false` for larger images
  final verified = await client.myEndpoint.verifyImageUpload(
    fileName: fileName,
  );

  if (verified) {
    return const Success();
  } else {
    return Failure();
  }
}
```

Endpoint methods:
```dart
Future<String?> getImageUploadDescription(
  Session session, {
  required String fileName,
}) async {

  final description = await session.storage.createDirectFileUploadDescription(
    storageId: 'public',
    path: fileName,
  );

  session.log('Created upload description for $fileName',
      level: LogLevel.debug);

  return description;
}

Future<bool> verifyImageUpload(
    Session session, {
    required String fileName,
  }) async {

    session.log('Verifying upload for $fileName', level: LogLevel.debug);

    return session.storage.verifyDirectFileUpload(
      storageId: 'public',
      path: fileName,
    );
}
```

I even tried to set `maxRequestSize: 10585760` in the server config, but I also know that file upload should support large files without increasing the default request size for the server.

**UPD.:** I'm using Flutter Web and just found this issue: https://github.com/serverpod/serverpod/issues/141

Any ideas?

**Comment by vipinkashyap** [comment]
I've been trying this too glad you could raise this issue. Any reason to limit file upload sizes ?

**Reply by estevez-dev** [reply]
I ended up using a self-hosted S3-compatible storage instead of the database. No solution for the database file storage for now.

**Comment by vlidholt** [comment, marked as answer]
By default, Serverpod sets the maximum request size to 512KB as a safeguard against spamming the server. If you need to upload larger files, you can increase this limit by setting the maxRequestSize key in your server's configuration file.
Here's an example of how to double the request size to 1MB in your development environment:

```yaml
# config/development.yaml
maxRequestSize: 1048576

apiServer:
  port: 8080
  publicHost: localhost
  publicPort: 8080
  publicScheme: http
```

The database file storage is not meant for storing large files and is not very efficient. It's much better to use GCP or AWS buckets. We may provide a better local file storage in the future.

**Reply by abdirahmanabdullahi** [reply]
Hello. How is this going??
